File: examples/python/e2b-pty/main.py
from __future__ import annotations

import os, sys

ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", ".."))
sys.path.insert(0, os.path.join(ROOT, "sdks", "python", "src"))

from sandboxer import Sandboxer, CreatePTYRequest  # noqa: E402
from sandboxer.errors import NotSupportedError  # noqa: E402


def main() -> None:
    key = os.environ.get("E2B_API_KEY")
    if not key:
        print("Set E2B_API_KEY", file=sys.stderr)
        sys.exit(1)
    cfg = {
        "api_key": key,
        "base_url": os.environ.get("E2B_API_BASE", "https://api.e2b.app"),
    }
    client = Sandboxer("e2b", cfg)
    sb, _info = client.create_sandbox()
    try:
        try:
            sb.create_pty(CreatePTYRequest())
        except NotSupportedError as e:
            print("PTY not supported by E2B provider in this SDK:", e)
    finally:
        sb.kill()
        client.close()


if __name__ == "__main__":
    main()

File: examples/go/e2b-pty/main.go
package main

import (
	"context"
	"errors"
	"fmt"
	"log"
	"os"

	"github.com/hyperterse/sandboxer/sdks/go"
	_ "github.com/hyperterse/sandboxer/sdks/go/providers"
)

func main() {
	key := firstNonEmpty(os.Getenv("E2B_API_KEY"), os.Getenv("SANDBOXER_API_KEY"))
	if key == "" {
		fmt.Fprintln(os.Stderr, "Set E2B_API_KEY (or SANDBOXER_API_KEY).")
		os.Exit(1)
	}
	ctx := context.Background()
	cfg := sandboxer.Config{Provider: sandboxer.ProviderE2B, APIKey: key}
	if b := os.Getenv("E2B_API_BASE"); b != "" {
		cfg.BaseURL = b
	}
	p, err := sandboxer.NewProvider(cfg)
	if err != nil {
		log.Fatal(err)
	}
	defer p.Close()

	sb, _, err := p.CreateSandbox(ctx, sandboxer.CreateSandboxRequest{Provider: sandboxer.ProviderE2B})
	if err != nil {
		log.Fatal(err)
	}
	defer sb.Kill(ctx)

	_, err = sb.CreatePTY(ctx, sandboxer.CreatePTYRequest{Rows: sandboxer.Ptr(24), Cols: sandboxer.Ptr(80)})
	if err != nil {
		if errors.Is(err, sandboxer.ErrNotSupported) {
			fmt.Println("CreatePTY not supported by this E2B driver build; see provider implementation.")
			return
		}
		log.Fatal(err)
	}
}

func firstNonEmpty(a, b string) string {
	if a != "" {
		return a
	}
	return b
}
